Issue on pod restore

I've seen this issue on the Jenkins in the last days related with the Geneve, today I replicated in my local environment: 

The data is: 

- Cilium: master
- Kubernetes version: 1.6
- Ginkgo Kubernetes base box. 

The ginkgo logs are the following: 

```
time="2018-01-09T16:17:34Z" level=debug msg="os.Hostname() returned" nodeName=k8s2
time="2018-01-09T16:17:34Z" level=info msg="     _ _ _"
time="2018-01-09T16:17:34Z" level=info msg=" ___|_| |_|_ _ _____"
time="2018-01-09T16:17:34Z" level=info msg="|  _| | | | | |     |"
time="2018-01-09T16:17:34Z" level=info msg="|___|_|_|_|___|_|_|_|"
time="2018-01-09T16:17:34Z" level=info msg="Cilium 0.13.90 5521815 Tue, 9 Jan 2018 12:48:15 +0100 go version go1.9 linux/amd64"
time="2018-01-09T16:17:34Z" level=info msg="clang (3.8.1) and kernel (4.9.68) versions: OK!"
time="2018-01-09T16:17:34Z" level=info msg="linking environment: OK!"
time="2018-01-09T16:17:35Z" level=info msg="bpf_requirements check: OK!"
time="2018-01-09T16:17:35Z" level=debug msg="bpf_features check: Some features may be limited:"
time="2018-01-09T16:17:35Z" level=debug msg="BPF/probes: Kernel config not found."
time="2018-01-09T16:17:35Z" level=debug msg="HAVE_LPM_MAP_TYPE: Your kernel doesn't support LPM trie maps for BPF, thus disabling CIDR policies. Recommendation is to run 4.11+ kernels."
time="2018-01-09T16:17:35Z" level=debug msg="HAVE_LRU_MAP_TYPE: Your kernel doesn't support LRU maps for BPF, thus switching back to using hash table for the cilium connection tracker. Recommendation is to run 4.10+ kernels."
time="2018-01-09T16:17:35Z" level=info msg="Waiting for etcd client to be ready..."
time="2018-01-09T16:17:35Z" level=info msg="Version of etcd endpoint OK" etcdEndpoint="http://k8s1:9732" version=3.1.0
time="2018-01-09T16:17:35Z" level=info msg="Using etcd as key-value store"
time="2018-01-09T16:17:35Z" level=info msg="Valid label prefix configuration:"
time="2018-01-09T16:17:35Z" level=info msg=" - :io.kubernetes.pod.namespace"
time="2018-01-09T16:17:35Z" level=info msg=" - !:io.kubernetes"
time="2018-01-09T16:17:35Z" level=info msg=" - !:.*kubernetes.io"
time="2018-01-09T16:17:35Z" level=info msg=" - !:pod-template-generation"
time="2018-01-09T16:17:35Z" level=info msg=" - !:pod-template-hash"
time="2018-01-09T16:17:35Z" level=info msg=" - !:controller-revision-hash"
time="2018-01-09T16:17:35Z" level=info msg=" - !:annotation.cilium.io/"
time="2018-01-09T16:17:35Z" level=info msg=" - !:annotation.cilium-identity"
time="2018-01-09T16:17:35Z" level=debug msg="Creating worker threads" count=4
time="2018-01-09T16:17:35Z" level=info msg="Waiting for k8s api-server to be ready..." subsys=k8s
time="2018-01-09T16:17:35Z" level=info msg="Connected to k8s api-server" ipAddr="https://192.168.36.11:6443" subsys=k8s
time="2018-01-09T16:17:35Z" level=info msg="Retrieved node information from kubernetes" nodeName=k8s2 subsys=k8s
time="2018-01-09T16:17:35Z" level=info msg="Received own node information from API server" ipAddr.ipv4=192.168.36.12 ipAddr.ipv6="<nil>" nodeName=k8s2 subsys=k8s
time="2018-01-09T16:17:35Z" level=info msg="Retrieved IPv4 allocation range for node. Using it for ipv4-range" node=k8s2 v4Prefix=10.10.1.0/24
time="2018-01-09T16:17:35Z" level=info msg="Retrieved IPv6 allocation range for node. Using it for ipv6-range" node=k8s2 v6Prefix="f00d::a0a:100:0:0/96"
time="2018-01-09T16:17:35Z" level=info msg="Automatically retrieved IP for node. Using it for ipv4-node" ipAddr=192.168.36.12 node=k8s2
time="2018-01-09T16:17:35Z" level=info msg="k8s mode: Allowing localhost to reach local endpoints"
time="2018-01-09T16:17:35Z" level=debug msg="Updating node annotations with node CIDRs" nodeName=k8s2 subsys=k8s v4Prefix=10.10.1.0/24 v6Prefix="f00d::a0a:100:0:0/96"
time="2018-01-09T16:17:35Z" level=info msg="Local node-name: k8s2"
time="2018-01-09T16:17:35Z" level=info msg="Node-IPv6: fc00::10ca:1"
time="2018-01-09T16:17:35Z" level=info msg="External-Node IPv4: 192.168.36.12"
time="2018-01-09T16:17:35Z" level=info msg="Internal-Node IPv4: 10.10.1.44"
time="2018-01-09T16:17:35Z" level=info msg="Cluster IPv6 prefix: f00d::/64"
time="2018-01-09T16:17:35Z" level=info msg="Cluster IPv4 prefix: 10.0.0.0/8"
time="2018-01-09T16:17:35Z" level=info msg="IPv6 node prefix: f00d::a0a:100:0:0/96"
time="2018-01-09T16:17:35Z" level=info msg="IPv6 allocation prefix: f00d::a0a:100:0:0/112"
time="2018-01-09T16:17:35Z" level=info msg="IPv4 allocation prefix: 10.10.1.0/24"
time="2018-01-09T16:17:35Z" level=debug msg="IPv6 router address: f00d::a0a:100:0:8ad6"
time="2018-01-09T16:17:35Z" level=debug msg="bpf: Setting tunnel endpoint" ipAddr=192.168.36.12 v4Prefix=10.10.1.0/24 v6Prefix="f00d::a0a:100:0:0/112"
time="2018-01-09T16:17:35Z" level=debug msg="Deferring node routes installation, host device not present yet"
time="2018-01-09T16:17:35Z" level=info msg="Removing mismatched map" file-path=cilium_policy_reserved_1
time="2018-01-09T16:17:35Z" level=info msg="Removing mismatched map" file-path=cilium_policy_reserved_2
time="2018-01-09T16:17:35Z" level=debug msg="writing configuration" file-path=netdev_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Checking local routes for conflicts..."
time="2018-01-09T16:17:38Z" level=debug msg="Ignoring route: no destination address" route="{Ifindex: 3 Dst: <nil> Src: <nil> Gw: 10.0.2.2 Flags: [] Table: 254}"
time="2018-01-09T16:17:38Z" level=debug msg="Considering route" route="{Ifindex: 3 Dst: 10.0.2.0/24 Src: 10.0.2.15 Gw: <nil> Flags: [] Table: 254}"
time="2018-01-09T16:17:38Z" level=debug msg="Considering route" route="{Ifindex: 4 Dst: 172.17.0.0/16 Src: 172.17.0.1 Gw: <nil> Flags: [] Table: 254}"
time="2018-01-09T16:17:38Z" level=debug msg="Considering route" route="{Ifindex: 2 Dst: 192.168.36.0/24 Src: 192.168.36.12 Gw: <nil> Flags: [] Table: 254}"
time="2018-01-09T16:17:38Z" level=info msg="Added L2 nexthop route" route="{Ifindex: 152 Dst: 10.10.1.44/32 Src: <nil> Gw: <nil> Flags: [] Table: 0}"
time="2018-01-09T16:17:38Z" level=info msg="Installed node route" route="{Ifindex: 152 Dst: 10.10.1.0/24 Src: 10.10.1.44 Gw: 10.10.1.44 Flags: [] Table: 0}"
time="2018-01-09T16:17:38Z" level=info msg="Added L2 nexthop route" route="{Ifindex: 152 Dst: f00d::a0a:100:0:8ad6/128 Src: <nil> Gw: <nil> Flags: [] Table: 0}"
time="2018-01-09T16:17:38Z" level=info msg="Installed node route" route="{Ifindex: 152 Dst: f00d::a0a:100:0:0/112 Src: fc00::10ca:1 Gw: f00d::a0a:100:0:8ad6 Flags: [] Table: 0}"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P PREROUTING ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P INPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P OUTPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P POSTROUTING ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N CILIUM_POST"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N DOCKER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-MARK-DROP"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-NODEPORTS"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-POSTROUTING"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SEP-H6QXRFVUSUGKBVUX"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SEP-JAZGIV2IL2Z3BV4H"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SEP-VWDM3Y7KJBIYBV7P"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SVC-ERIFXISQEP7F7OF4"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SVC-HN6JSMXPTE4HJDRY"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SVC-JSPWZOGCAB3SY23D"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SVC-NPX46M4PTMTKRN6Y"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SVC-TCOU7JCQXEZGVUNU"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A PREROUTING -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A OUTPUT -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A POSTROUTING -m comment --comment \"kubernetes postrouting rules\" -j KUBE-POSTROUTING"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A POSTROUTING -s 172.17.0.2/32 -d 172.17.0.2/32 -p tcp -m tcp --dport 5000 -j MASQUERADE"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A POSTROUTING ! -s 192.168.36.12/32 -o enp0s8 -j MASQUERADE"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A POSTROUTING -m comment --comment \"cilium-feeder: CILIUM_POST\" -j CILIUM_POST"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t nat -D POSTROUTING -m comment --comment cilium-feeder: CILIUM_POST -j CILIUM_POST]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A CILIUM_POST ! -s 10.10.1.44/32 -o cilium_host -m comment --comment \"cilium host->cluster masquerade\" -j SNAT --to-source 10.10.1.44"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t nat -D CILIUM_POST ! -s 10.10.1.44/32 -o cilium_host -m comment --comment cilium host->cluster masquerade -j SNAT --to-source 10.10.1.44]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A CILIUM_POST -s 10.10.1.0/24 ! -d 10.10.1.0/24 ! -o cilium_+ -m comment --comment \"cilium masquerade non-cluster\" -j MASQUERADE"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t nat -D CILIUM_POST -s 10.10.1.0/24 ! -d 10.10.1.0/24 ! -o cilium_+ -m comment --comment cilium masquerade non-cluster -j MASQUERADE]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A DOCKER -i docker0 -j RETURN"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A DOCKER ! -i docker0 -p tcp -m tcp --dport 5000 -j DNAT --to-destination 172.17.0.2:5000"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-MARK-DROP -j MARK --set-xmark 0x8000/0x8000"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-MARK-MASQ -j MARK --set-xmark 0x4000/0x4000"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-NODEPORTS -p tcp -m comment --comment \"default/test-nodeport:http\" -m tcp --dport 31615 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-NODEPORTS -p tcp -m comment --comment \"default/test-nodeport:http\" -m tcp --dport 31615 -j KUBE-SVC-JSPWZOGCAB3SY23D"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-POSTROUTING -m comment --comment \"kubernetes service traffic requiring SNAT\" -m mark --mark 0x4000/0x4000 -j MASQUERADE"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-H6QXRFVUSUGKBVUX -s 192.168.36.11/32 -m comment --comment \"default/kubernetes:https\" -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-H6QXRFVUSUGKBVUX -p tcp -m comment --comment \"default/kubernetes:https\" -m recent --set --name KUBE-SEP-H6QXRFVUSUGKBVUX --mask 255.255.255.255 --rsource -m tcp -j DNAT --to-destination 192.168.36.11:6443"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-JAZGIV2IL2Z3BV4H -s 10.10.0.51/32 -m comment --comment \"default/testds-service:\" -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-JAZGIV2IL2Z3BV4H -p tcp -m comment --comment \"default/testds-service:\" -m tcp -j DNAT --to-destination 10.10.0.51:80"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-VWDM3Y7KJBIYBV7P -s 10.10.0.51/32 -m comment --comment \"default/test-nodeport:http\" -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SEP-VWDM3Y7KJBIYBV7P -p tcp -m comment --comment \"default/test-nodeport:http\" -m tcp -j DNAT --to-destination 10.10.0.51:80"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES ! -s 10.10.0.0/16 -d 10.96.202.121/32 -p tcp -m comment --comment \"default/test-nodeport:http cluster IP\" -m tcp --dport 10080 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.202.121/32 -p tcp -m comment --comment \"default/test-nodeport:http cluster IP\" -m tcp --dport 10080 -j KUBE-SVC-JSPWZOGCAB3SY23D"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES ! -s 10.10.0.0/16 -d 10.96.0.1/32 -p tcp -m comment --comment \"default/kubernetes:https cluster IP\" -m tcp --dport 443 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.0.1/32 -p tcp -m comment --comment \"default/kubernetes:https cluster IP\" -m tcp --dport 443 -j KUBE-SVC-NPX46M4PTMTKRN6Y"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES ! -s 10.10.0.0/16 -d 10.96.0.10/32 -p udp -m comment --comment \"kube-system/kube-dns:dns cluster IP\" -m udp --dport 53 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.0.10/32 -p udp -m comment --comment \"kube-system/kube-dns:dns cluster IP\" -m udp --dport 53 -j KUBE-SVC-TCOU7JCQXEZGVUNU"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES ! -s 10.10.0.0/16 -d 10.96.0.10/32 -p tcp -m comment --comment \"kube-system/kube-dns:dns-tcp cluster IP\" -m tcp --dport 53 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.0.10/32 -p tcp -m comment --comment \"kube-system/kube-dns:dns-tcp cluster IP\" -m tcp --dport 53 -j KUBE-SVC-ERIFXISQEP7F7OF4"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES ! -s 10.10.0.0/16 -d 10.97.216.198/32 -p tcp -m comment --comment \"default/testds-service: cluster IP\" -m tcp --dport 80 -j KUBE-MARK-MASQ"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.97.216.198/32 -p tcp -m comment --comment \"default/testds-service: cluster IP\" -m tcp --dport 80 -j KUBE-SVC-HN6JSMXPTE4HJDRY"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -m comment --comment \"kubernetes service nodeports; NOTE: this must be the last rule in this chain\" -m addrtype --dst-type LOCAL -j KUBE-NODEPORTS"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SVC-HN6JSMXPTE4HJDRY -m comment --comment \"default/testds-service:\" -j KUBE-SEP-JAZGIV2IL2Z3BV4H"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SVC-JSPWZOGCAB3SY23D -m comment --comment \"default/test-nodeport:http\" -j KUBE-SEP-VWDM3Y7KJBIYBV7P"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SVC-NPX46M4PTMTKRN6Y -m comment --comment \"default/kubernetes:https\" -m recent --rcheck --seconds 10800 --reap --name KUBE-SEP-H6QXRFVUSUGKBVUX --mask 255.255.255.255 --rsource -j KUBE-SEP-H6QXRFVUSUGKBVUX"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SVC-NPX46M4PTMTKRN6Y -m comment --comment \"default/kubernetes:https\" -j KUBE-SEP-H6QXRFVUSUGKBVUX"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P PREROUTING ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P INPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P FORWARD ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P OUTPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P POSTROUTING ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P PREROUTING ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P OUTPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P INPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P FORWARD DROP"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-P OUTPUT ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N CILIUM_FORWARD"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N DOCKER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N DOCKER-ISOLATION"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N DOCKER-USER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-FIREWALL"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-N KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A INPUT -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A INPUT -j KUBE-FIREWALL"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -j DOCKER-USER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -j DOCKER-ISOLATION"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -o docker0 -j DOCKER"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -i docker0 ! -o docker0 -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -i docker0 -o docker0 -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A FORWARD -m comment --comment \"cilium-feeder: CILIUM_FORWARD\" -j CILIUM_FORWARD"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t filter -D FORWARD -m comment --comment cilium-feeder: CILIUM_FORWARD -j CILIUM_FORWARD]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A OUTPUT -m comment --comment \"kubernetes service portals\" -j KUBE-SERVICES"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A OUTPUT -j KUBE-FIREWALL"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A CILIUM_FORWARD -o cilium_host -m comment --comment \"cilium: clear masq bit for pkts to cilium_host\" -j MARK --set-xmark 0x0/0x4000"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t filter -D CILIUM_FORWARD -o cilium_host -m comment --comment cilium: clear masq bit for pkts to cilium_host -j MARK --set-xmark 0x0/0x4000]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A CILIUM_FORWARD -d 10.0.0.0/8 -o cilium_host -m comment --comment \"cilium: any->cluster forward accept\" -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t filter -D CILIUM_FORWARD -d 10.0.0.0/8 -o cilium_host -m comment --comment cilium: any->cluster forward accept -j ACCEPT]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A CILIUM_FORWARD -s 10.0.0.0/8 -i cilium_host -m comment --comment \"cilium: cluster->any forward accept\" -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Removing iptables rule" obj="[-t filter -D CILIUM_FORWARD -s 10.0.0.0/8 -i cilium_host -m comment --comment cilium: cluster->any forward accept -j ACCEPT]"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A DOCKER -d 172.17.0.2/32 ! -i docker0 -o docker0 -p tcp -m tcp --dport 5000 -j ACCEPT"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A DOCKER-ISOLATION -j RETURN"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A DOCKER-USER -j RETURN"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-FIREWALL -m comment --comment \"kubernetes firewall for dropping marked packets\" -m mark --mark 0x8000/0x8000 -j DROP"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.0.10/32 -p udp -m comment --comment \"kube-system/kube-dns:dns has no endpoints\" -m udp --dport 53 -j REJECT --reject-with icmp-port-unreachable"
time="2018-01-09T16:17:38Z" level=debug msg="Considering removing iptables rule" obj="-A KUBE-SERVICES -d 10.96.0.10/32 -p tcp -m comment --comment \"kube-system/kube-dns:dns-tcp has no endpoints\" -m tcp --dport 53 -j REJECT --reject-with icmp-port-unreachable"
time="2018-01-09T16:17:38Z" level=info msg="Setting sysctl net.core.bpf_jit_enable=1"
time="2018-01-09T16:17:38Z" level=info msg="Setting sysctl net.ipv4.conf.all.rp_filter=0"
time="2018-01-09T16:17:38Z" level=info msg="Setting sysctl net.ipv6.conf.all.disable_ipv6=0"
time="2018-01-09T16:17:38Z" level=debug msg="Adding local ip to endpoint map" ipAddr=10.10.1.44
time="2018-01-09T16:17:38Z" level=debug msg="Adding local ip to endpoint map" ipAddr=192.168.36.12
time="2018-01-09T16:17:38Z" level=debug msg="Adding local ip to endpoint map" ipAddr="fc00::10ca:1"
time="2018-01-09T16:17:38Z" level=debug msg="Adding local ip to endpoint map" ipAddr="f00d::a0a:100:0:8ad6"
time="2018-01-09T16:17:38Z" level=info msg="Restoring old cilium endpoints..."
time="2018-01-09T16:17:38Z" level=debug msg="Reading directory" endpointID=11004 file-path=/var/run/cilium/state/11004/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Found endpoint C header file" endpointID=11004 file-path=/var/run/cilium/state/11004/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Reading directory" endpointID=11005 file-path=/var/run/cilium/state/11005/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Found endpoint C header file" endpointID=11005 file-path=/var/run/cilium/state/11005/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Reading directory" endpointID=38061 file-path=/var/run/cilium/state/38061/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Found endpoint C header file" endpointID=38061 file-path=/var/run/cilium/state/38061/lxc_config.h
time="2018-01-09T16:17:38Z" level=debug msg="Restoring endpoint" endpointID=11004
time="2018-01-09T16:17:38Z" level=info msg="No workload could be associated with endpoint, removing endpoint" endpointID=38061
time="2018-01-09T16:17:38Z" level=info msg="No workload could be associated with endpoint, removing endpoint" endpointID=11005
panic: assignment to entry in nil map

goroutine 96 [running]:
github.com/cilium/cilium/pkg/endpoint.(*EndpointStatus).addStatusLog(0xc420376d00, 0xc420070780)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:748 +0x4f
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).logStatusLocked(0xc42014c3c0, 0x0, 0x0, 0x1c6703d, 0x11)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:1125 +0x156
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).SetStateLocked(0xc42014c3c0, 0x1c624d0, 0xd, 0x1c6703d, 0x11, 0xc420070730)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:1438 +0x134
main.(*Daemon).deleteEndpoint(0xc420353340, 0xc42014c3c0, 0x1)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/daemon/endpoint.go:345 +0x14a
main.(*Daemon).SyncState.func1(0xc420058201, 0xc420353340, 0xc42014c3c0, 0xc4200581c0, 0xc420058230)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/daemon/state.go:70 +0xa06
created by main.(*Daemon).SyncState
	/tmp/cilium-net-build/src/github.com/cilium/cilium/daemon/state.go:66 +0x1e4
```


This happens a few times in the Ginkgo testing over the lasts days: 

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/839/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/851/

--- a/pkg/endpoint/endpoint.go
+++ b/pkg/endpoint/endpoint.go
@@ -959,7 +959,11 @@ func ParseEndpoint(strEp string) (*Endpoint, error) {
 		return nil, fmt.Errorf("failed to parse base64toendpoint: %s", err)
 	}
 
-	if ep.Status == nil {
+	// We need to check for nil in Status, CurrentStatuses and Log, since in
+	// some use cases, status will be not nil and Cilium will eventually
+	// error/panic if CurrentStatus or Log are not initialized correctly.
+	// Reference issue GH-2477
+	if ep.Status == nil || ep.Status.CurrentStatuses == nil || ep.Status.Log == nil {
 		ep.Status = NewEndpointStatus()
 	}
 

